hack/gatewayapi-conformance.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# Get Gateway API CRD bundle-version
+BUNDLE_VERSION=$(oc get crds gateways.gateway.networking.k8s.io -ojson | jq -r '.metadata.annotations."gateway.networking.k8s.io/bundle-version"')
+
+if [[ "${BUNDLE_VERSION}" == "null" ]]; then
+    echo "Cannot find bundle-version annotation from Gateway API CRD"
+    exit 1
+fi
+echo "Gateway API CRD bundle-version: ${BUNDLE_VERSION}"
+
+echo "Create GatewayClass gateway-conformance"
+oc apply -f -<<EOF
+apiVersion: gateway.networking.k8s.io/v1
+kind: GatewayClass
+metadata:
+  name: gateway-conformance
+spec:
+  controllerName: openshift.io/gateway-controller
+EOF
+
+oc wait --for=condition=Accepted=true gatewayclass/gateway-conformance --timeout=300s
+
+echo "All gatewayclass status:"
+oc get gatewayclass -A
+
+echo "Go version: $(go version)"
+cd ../..
+mkdir -p kubernetes-sigs
+cd kubernetes-sigs
+rm -rf gateway-api
+
+# find the branch of gateway-api repo
+RELEASE_VERSION=${BUNDLE_VERSION/v/}
+RELEASE_VERSION=$(echo ${RELEASE_VERSION} | grep -oP "^\d+\.\d+")
```

Comment:
```suggestion
RELEASE_VERSION=$(\grep -oP "^\d+\.\d+" <<<"${BUNDLE_VERSION#v}")
```

`\grep`: prevents any grep aliases from impacting the output.
`${BUNDLE_VERSION#}`: removes the shortest match from the beginning of the variable.